pkg/repo/index.go

```diff
@@ -219,7 +219,7 @@ func (i IndexFile) WriteFile(dest string, mode os.FileMode) error {
 func (i *IndexFile) Merge(f *IndexFile) {
 	for _, cvs := range f.Entries {
 		for _, cv := range cvs {
-			if !i.Has(cv.Name, cv.Version) {
+			if !i.Has(cv.Name, "=="+cv.Version) {
```

Comment:
The `==` is not working here. `Has` is calling `Get` which is returning a `improper constraint` constraint error. It happens to be passing the test but not for the right reason.

The `Has` function needs to be rewritten so that it does a comparison without using constraint checking.

Reply:
I had wondered about that approach, but I was uneasy about the possible impact of making such a change. In particular, I note the explicit comment above that function around ignoring pre-release versions:

https://github.com/helm/helm/blob/0c7a1a5af5be49f78a61598ea449f4b50902d32b/pkg/repo/index.go#L154-L157

Looking around further, this seems like the core function used for finding versions in a Chart. I'm not familiar enough with the existing test cases to know whether modifications there could end up leading to some particularly adverse behaviour like unintentionally installing a prerelease version when the user didn't intend so.

Reply:
@joestringer Dug into this further. `Get` isn't entirely clear on the range vs exact version. The current form of the function has been around for years. Due to API compat via SemVer we can't really change that function.

`Has` notes that it looks for the exact chart version. Yet, it is not doing that. So, `Has` should look be changed to look for the exact version instead of calling `Get`.

The note on `Get` is when the version is empty and in that case this is the default behavior of the semver package. Since our case is when a version is supplied the comment does not relate.

Would you be up for making this change to `Has`?

Reply:
Done. I also realized that the `==` was invalid semver, it should have been `=` so it was previously only working due to `semver.NewConstraint()` always generating an error. That part is fixed.

Additionally, semver would determine that `1.9-dev` is semantically equivalent to `1.9.0` so to properly satisfy the documentation for `Has`, I added an extra direct string comparison of the versions to determine whether such versions should be merged.